include/aws/io/stream.h

```diff
@@ -38,17 +39,29 @@ struct aws_input_stream_vtable {
     int (*read)(struct aws_input_stream *stream, struct aws_byte_buf *dest);
     int (*get_status)(struct aws_input_stream *stream, struct aws_stream_status *status);
     int (*get_length)(struct aws_input_stream *stream, int64_t *out_length);
-    void (*destroy)(struct aws_input_stream *stream);
+    void (*impl_destroy)(struct aws_input_stream *stream);
 };
 
-struct aws_input_stream {
+struct aws_input_stream_options {
     struct aws_allocator *allocator;
-    void *impl;
     struct aws_input_stream_vtable *vtable;
+    void *impl;
```

Comment:
debatable/trivial: instead of passing in `void *impl`, what if we passed in `size_t sizeof_impl` and then new() did `aws_mem_acquire_many()` to create both the base-struct and impl in one go?

Reply:
hmmm, I feel it is a bit weird as the `impl` will still need a callback to destroy (To clean up resource besides of the memory allocation). So, the destroy callback will not need to clean up the memory seems like a bit weird to me.